### PR TITLE
Add channels and endpoints into the objectstore

### PR DESCRIPTION
**Pull request by mestery**
This commit adds an objectstore backend for channels and endpoints,
including Create/Delete/List methods for each as well. The objectstore
will be complete after this commit, and can cache all CRD objects
for NSM clients to utilize.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: plugins/objectstore/networkservice_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	nsTestNamespace = "default"
+)
+
+var networkservices *networkserviceStore
+
+func TestNetworkServiceStore(t *testing.T) {
+	networkservices := &networkserviceStore{}
+	networkservices.networkServicesStore = newNetworkServicesStore()
+
+	ns := netmesh.NetworkService{
```
Comment (sbezverk) [inline plugins/objectstore/networkservice_test.go]: I think line 38 and 42 should be combined.

File: plugins/objectstore/networkservicechannels_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	chTestNamespace = "default"
+)
+
+var channels *channelStore
+
+func TestChannelStore(t *testing.T) {
+	channels := &channelStore{}
+	channels.networkServiceChannelsStore = newNetworkServiceChannelsStore()
+
+	nsc := netmesh.NetworkService_NetmeshChannel{
```
Comment (sbezverk) [inline plugins/objectstore/networkservicechannels_test.go]: Same comment applies here.

File: plugins/objectstore/networkserviceendpoints_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	epTestNamespace = "default"
+)
+
+var endpoints *endpointStore
+
+func TestEndpointStore(t *testing.T) {
+	endpoints := &endpointStore{}
+	endpoints.networkServiceEndpointsStore = newNetworkServiceEndpointsStore()
+
+	nse := netmesh.NetworkServiceEndpoint{
```
Comment (sbezverk) [inline plugins/objectstore/networkserviceendpoints_test.go]: And here as well.

File: plugins/objectstore/networkservice_test.go (hunk 1 above)
Comment (mestery) [reply plugins/objectstore/networkservice_test.go]: Done.

File: plugins/objectstore/networkservicechannels_test.go (hunk 2 above)
Comment (mestery) [reply plugins/objectstore/networkservicechannels_test.go]: Done.

File: plugins/objectstore/networkserviceendpoints_test.go (hunk 3 above)
Comment (mestery) [reply plugins/objectstore/networkserviceendpoints_test.go]: Done.

**Comment by sbezverk** [comment]
LGTM

**Pull request merged**